src/devicewrapper.cpp

```diff
@@ -0,0 +1,66 @@
+/* ****************************************************************************
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+*
+*
+* This file is part of Microsoft R Host.
+*
+* Microsoft R Host is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Microsoft R Host is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Microsoft R Host.  If not, see <http://www.gnu.org/licenses/>.
+*
+* ***************************************************************************/
+
+#include "devicewrapper.h"
+#include "msvcrt.h"
+
+namespace rhost {
+    namespace grdevices {
+        devicewrapper::devicewrapper() {
+            _ver = R_GE_getVersion();
+        }
+
+
+        devicewrapper::devicewrapper(pDevDesc dd) {
+            _ver = R_GE_getVersion();
+            switch (_ver) {
+            case 11:
```

Comment:
>case [](start = 12, length = 4)

we support 10 as well.  it's the same struct as 11.  same for the 2 functions below.